packages/punchcard/lib/appsync/fragment.ts

```diff
@@ -0,0 +1,195 @@
+import { RecordShape, Shape, ShapeGuards } from '@punchcard/shape';
+import { TypeSpec, TypeSystem } from './type-system';
+
+export class ApiFragment<I extends TypeSystem = any> {
+  public static join<F1 extends ApiFragment>(
```

Comment:
Ughhh... these overloads are bumming me out. Hopefully [this](https://github.com/microsoft/TypeScript/issues/5453) happens soon :/